scaffolding-node/lib/scaffolding.sh

```diff
@@ -422,19 +423,25 @@ _detect_git() {
 }
 
 _detect_node() {
+
   if [[ -n "$scaffolding_node_pkg" ]]; then
     _node_pkg="$scaffolding_node_pkg"
     build_line "Detected Node.js version in Plan, using '$_node_pkg'"
   else
     local val
     val="$(_json_val package.json .engines.node)"
     if [[ -n "$val" ]]; then
+
       # TODO fin: Add more robust packages.json to Habitat package matching
       case "$val" in
         *)
-          local version_num
-          version_num=$(_extracted_version_number "$val")
-          _node_pkg="core/node/$version_num"
+          nearest_version_string=$(_nearest_version_on_builder "$val")
+          nearest_version="${nearest_version_string//\"}"
+
+          if [[ $nearest_version =~ (.*No compatible version of node found.*) ]]; then
+              exit_with "$nearest_version"
```

Comment:
Well, I thought `exit_with()` had a default exit code, but it looks it [doesn't](https://github.com/habitat-sh/habitat/blob/b78916ff19572cbc98cc3a239608c7327514e434/components/plan-build/bin/hab-plan-build.sh#L1397-L1416). Could you add some non-zero number as the second argument? There isn't any pattern that I've established for these numbers, which is why they look pretty random in other places.

Reply:
you bet!

Reply:
done!

Reply:
Er...now it's actually pushed!